src/systems/terrain_quad_tree.zig

```diff
@@ -87,15 +87,16 @@ const Mesh = struct {
     num_vertices: u32,
 };
 
+const invalid_index = std.math.maxInt(u32);
 const QuadTreeNode = struct {
     center: [2]f32,
     size: [2]f32,
     child_indices: [4]u32,
     mesh_lod: u32,
     patch_index: [2]u32,
     // TODO: Do not store these here when we implement streaming
-    heightmap_handle: gfx.TextureHandle,
-    splatmap_handle: gfx.TextureHandle,
+    heightmap_handle: ?gfx.TextureHandle,
```

Comment:
Just a question, handles tend to have an invalid state of their own (e.g. value 0). It can still be handy to use zig's optional-feature so it's fine either way, mostly just curious if it's strictly necessary?

Reply:
Oh, I didn't know about their invalid state. I believe I didn't implement the TextureHandle correcty and the texture pool these handle come from is not based on `zpool`. 

I can definitely update this if you want :)

Reply:
No don't think it matters too much in this case. And the explicitness is nice :)